lib/sparql_to_sw_solr/instance_solr_doc.rb

```diff
@@ -35,10 +37,23 @@ def solr_doc_hash
         return if CKEY_BLACKLIST.include?(@ckey)
         doc = init_doc
         add_doc_title_fields(doc)
+        add_doc_topic_fields(doc)
         doc
       end
     end
 
+    def sparql_prefixes
+      @sparql_prefixes ||= begin
+        bf = "PREFIX bf: <#{BF_NS}>"
+        mads = "PREFIX madsrdf: <#{MADSRDF}>"
+        [bf, mads].join("\n")
+      end
+    end
+
+    def sparql_query(query)
+      sparql.query(sparql_prefixes + query)
+    end
+
```

Comment:
We need to pair to discuss this, as it is much clearer and easier in my mind, but not yours.